core/embed/rust/src/ui/model_tt/component/homescreen/mod.rs

```diff
@@ -194,16 +203,40 @@ where
             let notification = self.get_notification();
 
             let res = get_image();
+            let data_ok = res.is_ok();
+            let mut show_default = true;
+
             if let Ok(data) = res {
+                if is_image_jpeg(data.as_ref()) {
+                    let mut input = BufferInput(data.as_ref());
+                    let mut hs_img = HomescreenJpeg::new(&mut input);
+                    homescreen(
+                        &mut hs_img,
+                        &[text],
+                        notification,
+                        self.paint_notification_only,
+                    );
+                    show_default = false;
+                } else if data_ok && is_image_toif(data.as_ref()) {
```

Comment:
IMO `data_ok` is always true since we're inside `if let Ok(...) = res {` block and can be removed.

Reply:
https://github.com/trezor/trezor-firmware/commit/efad3596e8d2fc3c208e13384990bda439334289